qiskit/utils/deprecation.py

```diff
@@ -17,10 +17,15 @@
 from typing import Type
 
 
-def deprecate_arguments(kwarg_map, category: Type[Warning] = DeprecationWarning):
+def deprecate_arguments(
+    kwarg_map, category: Type[Warning] = DeprecationWarning, docstring_version=None
+):
     """Decorator to automatically alias deprecated argument names and warn upon use."""
 
     def decorator(func):
+        if docstring_version and kwarg_map:
+            _extend_docstring(func, docstring_version, kwarg_map)
+
```

Comment:
Logically this is a bit weird in that you mutate the docstring of the inner function, immediately before replacing that object with a new one.  It works because `functools.wraps` copies the docstring over, but just saying it's a bit unusual to do it this way round.

Reply:
True. Changing as out-place in b87339343e80cc3c53fac2bfd60484e257334c51